test/unit/test_Ites.cc

```diff
@@ -0,0 +1,188 @@
+//
+// Created by Antti on 04.08.20.
+//
+#include <gtest/gtest.h>
+#include <Logic.h>
+#include <LRALogic.h>
+#include "../../src/itehandler/IteToSwitch.h"
+
+class LogicIteTest: public ::testing::Test {
+public:
+    Logic logic;
+    LogicIteTest() : logic{} {}
+};
+
+class LRAIteTest: public ::testing::Test {
+public:
+    LRALogic logic;
+    LRAIteTest() : logic{} {}
+};
+
+class IteManagerTest: public ::testing::Test {
+public:
+    LRALogic logic;
+    SRef lrasort;
+
+    IteManagerTest() : lrasort(logic.getSort_num()) {}
+
+    void printTopLevelSwitches(IteToSwitch &iteManager) {
+        PTRef tr = logic.getTerm_true();
+        tr = iteManager.conjoin(tr);
+        std::cout << logic.pp(tr) << endl;
+    }
+};
+
+TEST_F(LogicIteTest, test_UFIte) {
+    SRef ufsort = logic.declareSort("U", nullptr);
+
+    PTRef x = logic.mkVar(ufsort, "x");
+    PTRef y = logic.mkVar(ufsort, "y");
+    PTRef cond = logic.mkEq(x, y);
+
+    vec<PTRef> args;
+    args.push(cond);
+    args.push(x);
+    args.push(y);
+    PTRef ite = logic.mkIte(args);
+    std::cout << logic.pp(ite) << endl;
+
+    args.clear();
+    args.push(logic.getTerm_true());
+    args.push(x);
+    args.push(y);
+    ite = logic.mkIte(args);
+    ASSERT_EQ(ite, x);
+
+    args.clear();
+    args.push(logic.getTerm_false());
+    args.push(x);
+    args.push(y);
+    ite = logic.mkIte(args);
+    ASSERT_EQ(ite, y);
+
+}
+
+TEST_F(LogicIteTest, test_BoolIte) {
+    SRef boolsort = logic.getSort_bool();
+
+    PTRef x = logic.mkVar(boolsort, "x");
+    PTRef y = logic.mkVar(boolsort, "y");
+    PTRef cond = logic.mkEq(x, y);
+
+    vec<PTRef> args;
+    args.push(cond);
+    args.push(x);
+    args.push(y);
+    PTRef ite = logic.mkIte(args);
+    std::cout << logic.pp(ite) << endl;
```

Comment:
It would be good if these tests would contain some assert, something to test that the terms are created correctly.